components/emoji/components/emoji_list.jsx

```diff
@@ -16,6 +16,10 @@ import LoadingScreen from 'components/loading_screen.jsx';
 
 import EmojiListItem from './emoji_list_item.jsx';
 
+function deleteEmoji(emoji) {
```

Comment:
Why not inside the class?

Reply:
My reasoning was because it is a static method that gets re-instantiated for every instance of the class. Outside the class it gets instantiated at runtime, and reused for every class. Perhaps it is a premature optimization though and shouldn't have been done though without testing.
It is more of a style issue than a major change.

Reply:
Do you want this reverted, or is it good this way? Waiting on this to be resolved to mark ready for merge

Reply:
If this is more of style/pattern, then I would prefer to have it inside the class like what we're doing elsewhere.

Reply:
I will change this back, because I am only 1/5 on it.

 However, I am curious how decisions like this are made? Is it simply "how we always did it", or is there a proper style guide or decision making framework to guide these calls? I simply ask because I would hate for us to make poor decisions habitually just because they are the same as other outdated or poor decisions.

Reply:
I think you may open up your idea or opinion in any of the discussion channel like developer channel, or might good to open a topic in developer meeting so you will hear instant feedback/discussion/agreement among developers.

Reply:
great idea. I will look into that.